### PR TITLE
test(recordings): Add recording options itests

### PR DESCRIPTION
**Pull request by jan-law**
Related #474 

Tests successful scenarios for the `TargetRecordingOptionsGetHandler`, `TargetRecordingOptionsPatchHandler`, and the `TargetRecordingOptionsListGetHandler`.

### CODE REVIEW

**Comment by andrewazores** [comment]
Tests look good. Do you still observe #263? Either way, I think it would be worth writing some tests to capture that behaviour.

**Comment by jan-law** [comment]
> Do you still observe #263?

Yes, new recordings are created with `toDisk=true`.
Should recordings write to disk by default? If you'd like, I could change the default options to return `toDisk=true`.

**Comment by andrewazores** [comment]
The default should be whatever the target JVM says it is, which according to [docs](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Recording.html#isToDisk()), seems it should be `true`. #263 seems to reflect a bug in JMC code, then, where the recording options map contains `false` when it would be expected to be `true` or perhaps `null` (unset, as the docs say).

File: src/test/java/itest/TargetRecordingOptionsIT.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+                                        ar.result().statusCode(), Matchers.equalTo(200));
+                                MatcherAssert.assertThat(
+                                        ar.result().getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                        Matchers.equalTo(HttpMimeType.JSON.mime()));
+                                getResponse.complete(ar.result().bodyAsJsonObject());
+                            }
+                        });
+
+        JsonObject expectedGetResponse =
+                new JsonObject(Map.of("maxAge", 0, "toDisk", false, "maxSize", 0));
```
Comment (andrewazores) [inline src/test/java/itest/TargetRecordingOptionsIT.java:158]: Could you put a `FIXME` here with a link to #263? That's still something we should probably investigate further, although I am fairly convinced that it's just a JMC bug with how those OptionDescriptors are created or how the toDisk setting is read from the JVM.

**Review by andrewazores — COMMENTED** [review]
Looks good otherwise.

**Review by andrewazores — APPROVED** [review]

**Comment by jan-law** [comment]
`RecordingWorkflowIT` failed at the part where it expects the in-memory jfr file to be larger than the saved jfr file. Why would it fail this time but pass most of the time? I'm re-running the tests to see what happens
```
Error:  Failures: 
Error:    RecordingWorkflowIT.testWorkflow:222 
Expected: a value greater than <9919>
but: <8305> was less than <9919>
```

**Comment by jan-law** [comment]
Hmm, now `AutoRulesIT` is failing because I forgot to reset the default recording options. Let me fix that

```
Error:  Failures: 
Error:    AutoRulesIT.testNewContainerHasRuleApplied:285 
Expected: <0>
     but: was <1000>
```

**Comment by andrewazores** [comment]
> `RecordingWorkflowIT` failed at the part where it expects the in-memory jfr file to be larger than the saved jfr file. Why would it fail this time but pass most of the time? I'm re-running the tests to see what happens
> 
> ```
> Error:  Failures: 
> Error:    RecordingWorkflowIT.testWorkflow:222 
> Expected: a value greater than <9919>
> but: <8305> was less than <9919>
> ```

Hmm. It sounds like it could simply be that the in-memory ring buffers filled up and some older event data was dropped in between the two recordings being retrieved, assuming that the recording isn't somehow configured with a maximum event age.

It's not a good fix, but changing the template in that test from `ALL` to just `Continuous` should cut down the number of events recorded and hopefully work around this edge case. Could you file a bug for this failure with a link to the CI log and whatever other findings you may have?

**Comment by andrewazores** [comment]
> Hmm, now `AutoRulesIT` is failing because I forgot to reset the default recording options. Let me fix that
> 
> ```
> Error:  Failures: 
> Error:    AutoRulesIT.testNewContainerHasRuleApplied:285 
> Expected: <0>
>      but: was <1000>
> ```

Aha - that would do it. `maxSize=1000` in your 4th test case would be likely to cause the behaviour I described in the last comment, or at least cause it to occur much earlier/more frequently.

File: src/test/java/itest/TargetRecordingOptionsIT.java [hunk 2]
```diff
@@ -63,6 +64,35 @@
...
+
+        MatcherAssert.assertThat(getResponse.get(), Matchers.equalTo(expectedGetResponse));
```
Comment (andrewazores) [inline src/test/java/itest/TargetRecordingOptionsIT.java:93]: I think we can skip the test assertion in this `AfterAll`, and the `BeforeAll` too if you add it.

Comment (jan-law) [reply src/test/java/itest/TargetRecordingOptionsIT.java:93]: Sounds good

File: src/test/java/itest/TargetRecordingOptionsIT.java [hunk 3]
```diff
@@ -63,6 +64,35 @@
             String.format("/api/v2/targets/%s/recordingOptionsList", SELF_REFERENCE_TARGET_ID);
     static final String RECORDING_NAME = "test_recording";
 
+    @AfterAll
+    static void resetDefaultRecordingOptions() throws Exception {
+        CompletableFuture<JsonObject> getResponse = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.add("maxAge", "0");
+        form.add("toDisk", "false");
+        form.add("maxSize", "0");
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply src/test/java/itest/TargetRecordingOptionsIT.java:74]: Actually - I guess the `toDisk` bug with 263 would bite us here, so at least that one would need to be overridden/hardcoded as `true` again. Nevermind then.

**Review by andrewazores — APPROVED** [review]
